src/sponge/absorb.rs

```diff
@@ -227,6 +228,16 @@ impl Absorb for isize {
     }
 }
 
+impl Absorb for String {
+    fn to_sponge_bytes(&self, dest: &mut Vec<u8>) {
+        dest.extend_from_slice(self.as_bytes())
```

Comment:
Presumably this absorbs the length too?

Reply:
I can also implement `AbsorbWithLength` for `String` and leave this as is, wdyt?
Then it's just the question of whether the caller knows about the trait `AbsorbWithLength`, which arguably is not so obvious since most types only implement pure `Absorb`.

Reply:
I've added the length by default to `to_sponge_bytes`.

I left `to_sponge_field_elements` as-is. It already prevents the scenario you mentioned: each substring gets mapped to a new field element, whereas a concat string gets mapped to a single element, so the resulting vectors even have different lengths.